src/atoms/cc-badge.js

```diff
@@ -0,0 +1,145 @@
+import { css, html, LitElement } from 'lit-element';
+import { defaultThemeStyles } from '../styles/default-theme.js';
+
+/**
+ * @typedef {import('./types.js').BadgeIntent} BadgeIntent
+ * @typedef {import('./types.js').BadgeWeight} BadgeWeight
+ */
+
+/**
+ * A component to highlight a small chunk of text.
+ *
+ * @cssdisplay inline-flex
+ */
+export class CcBadge extends LitElement {
+  static get properties () {
+    return {
+      circle: { type: Boolean, reflect: true },
+      iconAlt: { type: String, attribute: 'icon-alt' },
+      iconSrc: { type: String, attribute: 'icon-src' },
+      intent: { type: String, reflect: true },
+      weight: { type: String, reflect: true },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {Boolean} Sets the badge to a bubble style. Should only be used to display 1 or 2 digits figures. */
+    this.circle = false;
+
+    /** @type {String|null} Sets the `alt` attribute value on the `<img>` tag. Only use if the image conveys additional info compared to surrounding text. */
+    this.iconAlt = null;
+
+    /** @type {String|null} Sets the icon displayed on the left of the text inside the badge. */
+    this.iconSrc = null;
+
+    /** @type {BadgeIntent} Sets the accent color used for the badge. */
+    this.intent = 'neutral';
+
+    /** @type {BadgeWeight} Sets the style of the badge depending on how much one wants it to stand out. */
+    this.weight = 'dimmed';
+  }
+
+  render () {
+    return html`
+      ${this.iconSrc != null ? html`
```

Comment:
considering we use a slot, should we also slot the image ? :roll_eyes: 

I guess right now it matches the `cc-button` behavior (slot for text, prop for image) so I'm okay with it :+1: